druid/src/widget/focus.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020 The druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A focus widget.
+
+use druid::widget::prelude::*;
+
+use druid::{commands, Data, FocusNode, HotKey, KbKey, Point, Rect, SysMods, Widget, WidgetPod};
+
+#[derive(Debug, Clone)]
+pub(crate) struct FocusController {
```

Comment:
A rule of thumb: anything that druid's built-in widgets use in their implementations should be public; so if this is used in `TextBox` we should expose it, so that other people can write their own textboxes.

Reply:
Okay, got it.